### PR TITLE
MDS 4199 Ability to view imported IRT

### PR DESCRIPTION
**Pull request by areyeslo**
## Objective 

https://bcmines.atlassian.net/browse/MDS-4199

Ability to view an imported IRT in step 3:
image

## Additional Information / Context 
This work finishes the basic implementation of IRT. Following tickets would add functionality.

### CODE REVIEW

File: services/minespace-web/src/components/Forms/projects/informationRequirementsTable/InformationRequirementsTableForm.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  deepMergeById = (r1, r2) =>
+    r1.map(({ requirement_guid, sub_requirements, ...rest }) => ({
+      requirement_guid,
+      ...rest,
+      ...r2.find((i) => i.requirement_guid === requirement_guid),
+      sub_requirements: this.deepMergeById(sub_requirements, r2),
+    }));
+
+  render() {
+    const renderTabComponent = (tab, tabIndex, requirements) =>
```
Comment (BenHarrington) [inline services/minespace-web/src/components/Forms/projects/informationRequirementsTable/InformationRequirementsTableForm.js]: Is there anyway we could just pass in the key to this common component to clean up this function?

File: services/minespace-web/src/components/Forms/projects/informationRequirementsTable/ReviewSubmitInformationRequirementsTable.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  <>
+    <Col span={24}>
+      <span id="description" style={DescriptionTextSytle(sub_requirement.step)}>
+        {`${sub_requirement.step} ${sub_requirement.description}`}
+      </span>
+
+      {sub_requirement.required === undefined && sub_requirement.methods === undefined ? null : (
+        <>
+          <p>
+            <Checkbox checked={sub_requirement.required.toLowerCase() === "true"}>
```
Comment (BenHarrington) [inline services/minespace-web/src/components/Forms/projects/informationRequirementsTable/ReviewSubmitInformationRequirementsTable.js]: Is there a reason the boolean is a string here?

**Review by BenHarrington — COMMENTED** [review]
Great work Arturo, this feature looks awesome!

File: services/minespace-web/src/components/pages/Project/InformationRequirementsTablePage.js [hunk 3]
```diff
@@ -109,29 +173,59 @@ export class InformationRequirementsTablePage extends Component {
...
+
   handleFetchData = () => {
-    const { projectGuid, tab } = this.props.match?.params;
+    const { projectGuid } = this.props.match?.params;
 
     return this.props
       .fetchProjectById(projectGuid)
-      .then(() => this.setState({ isLoaded: true, activeTab: tab }));
+      .then(() => this.props.fetchRequirements())
+      .then(() => this.setState({ isLoaded: true }));
```
Comment (BenHarrington) [inline services/minespace-web/src/components/pages/Project/InformationRequirementsTablePage.js:201]: This seems like a redundant setState, is there a reason we are doing this again?

File: services/minespace-web/src/components/Forms/projects/informationRequirementsTable/ReviewSubmitInformationRequirementsTable.js (hunk 2 above)
Comment (areyeslo) [reply services/minespace-web/src/components/Forms/projects/informationRequirementsTable/ReviewSubmitInformationRequirementsTable.js]: Fixing response model in backend to return Boolean.

File: services/minespace-web/src/components/Forms/projects/informationRequirementsTable/ReviewSubmitInformationRequirementsTable.js [hunk 4]
```diff
@@ -28,11 +28,11 @@ const ReviewSubmitInformationRequirementsTable = ({ requirements }) => {
 const DescriptionTextSytle = (step) => {
```
Comment (BenHarrington) [inline services/minespace-web/src/components/Forms/projects/informationRequirementsTable/ReviewSubmitInformationRequirementsTable.js]: Just caught this typo here, other than that everything looks good to go!

**Review by BenHarrington — APPROVED** [review]

**Pull request merged**